src/lib/Libifl/ifl_util.c

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (C) 1994-2020 Altair Engineering, Inc.
+ * For more information, contact Altair at www.altair.com.
+ *
+ * This file is part of both the OpenPBS software ("OpenPBS")
+ * and the PBS Professional ("PBS Pro") software.
+ *
+ * Open Source License Information:
+ *
+ * OpenPBS is free software. You can redistribute it and/or modify it under
+ * the terms of the GNU Affero General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * OpenPBS is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Commercial License Information:
+ *
+ * PBS Pro is commercially licensed software that shares a common core with
+ * the OpenPBS software.  For a copy of the commercial license terms and
+ * conditions, go to: (http://www.pbspro.com/agreement.html) or contact the
+ * Altair Legal Department.
+ *
+ * Altair's dual-license business model allows companies, individuals, and
+ * organizations to create proprietary derivative works of OpenPBS and
+ * distribute them - whether embedded or bundled with other software -
+ * under a commercial license agreement.
+ *
+ * Use of Altair's trademarks, including but not limited to "PBS™",
+ * "OpenPBS®", "PBS Professional®", and "PBS Pro™" and Altair's logos is
+ * subject to Altair's trademark licensing policies.
+ */
+
+#include "pbs_ifl.h"
+#include "libpbs.h"
+
+/**
+ * @brief
+ *	-returns the server name.
+ *
+ * @param[in] server - server name
+ * @param[out] server_name - server name
+ * @param[in] port - port number
+ *
+ * @return	string
+ * @retval	servr name	success
+ *
+ */
+char *
+PBS_get_server(char *server, char *server_name, unsigned int *port)
+{
+	int   i;
+	char *pc;
+	unsigned int dflt_port = 0;
+	char *p;
+
+	for (i=0;i<PBS_MAXSERVERNAME+1;i++)
+		server_name[i] = '\0';
+
+	if (dflt_port == 0)
+		dflt_port = pbs_conf.batch_service_port;
+
+	/* first, get the "net.address[:port]" into 'server_name' */
+
+	if ((server == NULL) || (*server == '\0')) {
+		if ((p=pbs_default()) == NULL)
+			return NULL;
+		strcpy(server_name, p);
+	} else {
```

Comment:
I prefer else to be enclosed with curly braces if the corresponding if is enclosed.

Reply:
Most of our code does not have braces for single line conditional blocks. If you leave it, someone who will work on this code will have to remove it. So better keep it the same as the majority of instances. 

Also, its documented in our coding standards: "You don't need braces for single statements"
https://openpbs.atlassian.net/wiki/spaces/DG/pages/1481670667/PBS+Pro+Coding+Standards

Reply:
the guideline also says "Use parentheses anywhere it makes it easier for the reader"
It is hard to notice the else after a lengthy if block. So I prefer it to have the braces which will bring attention to the else.

Reply:
I understand your concern. I can live with you not removing it but also know that since it is in our guidelines, someone who will work on this code next will remove it eventually. It's not a big if block and with correct indentation, I don't think it's hard to read.

Reply:
I definitely see the benefit of having braces when the corresponding if/else clause is lengthy. I see it is hard to apply such a rule as it is intuitive and depends on the number of lines and the width as well. I am leaving it this way as long as you don't see it is less readable with the braces on.

Reply:
I will not block the PR on this. But it will stay there till someone sees and removes the braces since it doesn't meet our guidelines.